src/meta/src/hummock/compaction_schedule_policy.rs

```diff
@@ -269,20 +294,59 @@ impl ScoredPolicy {
     }
 }
 
+impl ScoredPolicy {
+    fn fetch_idle_compactor(
+        &self,
+        compactor_assigned_task_num: &HashMap<HummockContextId, u64>,
+    ) -> Option<Arc<Compactor>> {
+        for compactor in self.score_to_compactor.values() {
+            let running_task = *compactor_assigned_task_num
+                .get(&compactor.context_id())
+                .unwrap_or(&0);
+            if running_task < compactor.max_concurrent_task_number() {
+                return Some(compactor.clone());
+            }
+        }
+        None
+    }
+}
+
 impl CompactionSchedulePolicy for ScoredPolicy {
     fn next_idle_compactor(
-        &self,
+        &mut self,
         compactor_assigned_task_num: &HashMap<HummockContextId, u64>,
     ) -> Option<Arc<Compactor>> {
+        if let Some(compactor) = self.fetch_idle_compactor(compactor_assigned_task_num) {
+            if let CompactorState::Idle(_) = &self.state {
+                // do not change state.
+            } else {
+                self.state = CompactorState::Idle(Instant::now());
+            }
+            return Some(compactor);
+        }
+        match self.state {
+            CompactorState::Idle(_) => {
+                self.state = CompactorState::Burst(Instant::now());
+            }
+            CompactorState::Burst(last_burst) => {
+                if last_burst.elapsed().as_secs() > MAX_BURST_TIME {
+                    self.state = CompactorState::Busy;
+                    return None;
+                }
+            }
+            CompactorState::Busy => {
+                return None;
+            }
+        }
         for compactor in self.score_to_compactor.values() {
-            if *compactor_assigned_task_num
+            let running_task = *compactor_assigned_task_num
                 .get(&compactor.context_id())
-                .unwrap_or(&0)
-                < compactor.max_concurrent_task_number()
-            {
+                .unwrap_or(&0);
+            if running_task < 2 * compactor.max_concurrent_task_number() {
```

Comment:
In this Pr, the limit of running_task for per compactor is `max_concurrent_task_number * 2` when Scheduler in `Burst` state, right ?